### DISCUSSION TITLE
Got conn.Authenticate error when replicated with restic based mode

### DISCUSSION THREAD

**Discussion by sulaimanyu** [backube/volsync#1241] [General]
I created restic repo with following statements:

apiVersion: v1
kind: Secret
metadata:
  name: pvc-backup
type: Opaque
stringData:
  RESTIC_REPOSITORY: swift:pvc-backup:/pvc-backup
  RESTIC_PASSWORD: restic1234
  OS_AUTH_URL: "https://auth_url:5000/v3"
  OS_PROJECT_NAME: "projectname"
  OS_USER_DOMAIN_NAME: "Default"
  OS_PROJECT_DOMAIN_ID: "default"
  OS_USERNAME: "username"
  OS_PASSWORD: "pwd"
  OS_REGION_NAME: "regionOne"

and create replicationsource to replicate:

apiVersion: volsync.backube/v1alpha1
kind: ReplicationSource
metadata:
  name: backup-pvc
spec:
  sourcePVC: my-pvc
  trigger:
    schedule: "0 */4 * * *"
  restic:
    pruneIntervalDays: 30
    repository: pvc-backup
    retain:
      hourly: 6
      daily: 4
    copyMethod: Clone

But the replication can not proceed with error:

Fatal: unable to open repository at swift:pvc-backup:/pvc-backup: conn.Authenticate: Response didn't have storage url and auth token 

ERROR: failure checking existence of repository 

It seems that the restic repo can not be created. 

But if I use restic cli to init this repo with all the same parameters. the restic repo can be created successfully.

Any suggestion for this issue?

Thanks

**Comment by tesshuflower** [comment]
I don't have any direct experience with swift storage - but it's interesting you're able to connect with the restic cli directly.

Is the OS_AUTH_URL you're using accessible from within the cluster?

**Reply by sulaimanyu** [reply]
Yes, I retrieved OS_AUTH_URL from the openstack RC file

**Reply by tesshuflower** [reply]
Not sure what else to suggest - you will need to be able to access from within the cluster (and from the namespace your ReplicationSource is in).

A couple thoughts if you want to debug further:
- Is there anything in the logs for the mover pod for the replicationsource?
- When the mover pod for the replicationsource runs, you can check that it has the proper env vars set (and all to your expected values)
- You could try running restic yourself from the namespace in the cluster to confirm it's accessible (for example you could run a pod with the volsync img itself and run restic from there).

something like:
```
kubectl run -n mynamespace mypodname --image=quay.io/backube/volsync:0.9.1  --tty -i --command /bin/bash
```
